### PR TITLE
Keep symbols for mlt

### PR DESCRIPTION
**Pull request by acrrd**
When using the mlt endpoint we don't need to strip the symbols. Not doing it will return more results in some case.

### CODE REVIEW

File: discovery_engine_core/ai/ai/src/coi/key_phrase.rs [hunk 1]
```diff
@@ -213,7 +213,6 @@ fn unify(
     let mut candidates = candidates
         .iter()
         .filter_map(|candidate| {
-            let candidate = clean_query(candidate);
```
Comment (rustonaut) [inline discovery_engine_core/ai/ai/src/coi/key_phrase.rs:216]: shouldn't we make this conditional based on weather the mlt endpoint is used?

Comment (acrrd) [reply discovery_engine_core/ai/ai/src/coi/key_phrase.rs:216]: We have no idea if that is used or not here. There is no plan to revert the use of mlt.
We can always do `clean_query` when generating the filter and that is probably the correct place where to do it.

**Comment by rustonaut** [comment]
@acrrd  not sure if you didn't notice or if you had other things to do, but there are still some tests failing in `key_phrase.rs` due to the change.

They mainly test indirectly if the cleaning works so I guess some of the failing ones could be just removed.

Also I assume we did make sure to properly validate/sanitize input in the mlt endpoint.(?)

**Comment by acrrd** [comment]
I'm waiting for the kps split to be merged back since it modifies something here. Also we still need to escape single e double quote. So I think I'll close this and have another task.

**Comment by acrrd** [comment]
We need more understanding of what symbols are ok and what are not for this endpoint.

**Pull request closed without merging**